lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.cpp

```diff
@@ -95,6 +95,54 @@ float ACS711_Current_Bus::adc_to_current_15A(float adc_value, float adc_fsr, flo
 }
 
 int16_t ACS711_Current_Bus::read(uint8_t bus, uint8_t mux) { return curr_buffer_[bus][mux]; }
+#else
+ADS1119Configuration ADS1119_Current_Bus::configurations[BUSES][MUXES] = {};
+ADS1119 ADS1119_Current_Bus::ads1 = ADS1119(ads1_addr);
+ADS1119 ADS1119_Current_Bus::ads2 = ADS1119(ads2_addr);
+
+void ADS1119_Current_Bus::init_ads1119() {
+  //init all channels to same general config, but with different mux config. 
+  //different configs for both chips for futureproofing, not because they're any different
+  for (int i = 0; i < BUSES; ++i) {
+    configurations[i][0].mux = ADS1119MuxConfiguration::positiveAIN0negativeAGND;
+    configurations[i][1].mux = ADS1119MuxConfiguration::positiveAIN1negativeAGND;
+    configurations[i][2].mux = ADS1119MuxConfiguration::positiveAIN2negativeAGND;
+    configurations[i][3].mux = ADS1119MuxConfiguration::positiveAIN3negativeAGND;
+
+    for (int j = 0; j < MUXES; ++j) {
+      configurations[i][j].gain = ADS1119Configuration::Gain::one;
+      configurations[i][j].dataRate = ADS1119Configuration::DataRate::sps330;
+      configurations[i][j].conversionMode = ADS1119Configuration::ConversionMode::continuous;
+      configurations[i][j].voltageReference = ADS1119Configuration::VoltageReferenceSource::external;
+      configurations[i][j].externalReferenceVoltage = 3.3;
+    }
+  }
+
+  ads1.begin();
+  ads1.reset();
+  
+  ads2.begin();
+  ads2.reset();
+}
+
+float ADS1119_Current_Bus::read(uint8_t bus, uint8_t mux) {
+  switch (bus) {
+    case 0:
+    return ADS1119_Current_Bus::adc_to_current_31A(ads1.readVoltage(configurations[0][mux]));
+    case 1: 
+      return ADS1119_Current_Bus::adc_to_current_31A(ads2.readVoltage(configurations[1][mux]));
+    default:
+      //we only have two muxes
+      return 69.420;
+  }
+}
+
+float ADS1119_Current_Bus::adc_to_current_31A(float adc_value, float adc_fsr, float vcc) {
```

Comment:
would recommend adding the current sensor chip name as a comment around here for clarity (the conversions are specific to each chip and can easily be incorrectly referenced)